tensorboard/backend/application.py

```diff
@@ -437,7 +437,24 @@ def _serve_plugins_listing(self, request):
 
       es_module_handler = plugin_metadata.es_module_path
       element_name = plugin_metadata.element_name
-      if element_name is not None and es_module_handler is not None:
+      is_ng_component = plugin_metadata.is_ng_component
+      if is_ng_component:
+        if element_name is not None:
+          logger.error(
+              'Plugin %r declared as both Angular built-in and legacy; skipping',
```

Comment:
I would be surprised if plugin registration does not happen because I specified both `is_ng_component` and `element_name` all together. Rather than `logger.error`, if this is that critical, I would want to see hard failure.

Also, a short test would be nice.

Reply:
Done making these cases hard `ValueError`s. Done adding tests.